recipes-bsp/rpi-u-boot-scr/files/boot.cmd.in

```diff
@@ -1,5 +1,31 @@
 fdt addr ${fdt_addr} && fdt get value bootargs /chosen bootargs
 fatload mmc 0:1 ${kernel_addr_r} @@KERNEL_IMAGETYPE@@
+
 if test ! -e @@BOOT_MEDIA@@ 0:1 uboot.env; then saveenv; fi;
-if env exists rpipart;then setenv bootargs "${bootargs} root=/dev/mmcblk0p${rpipart}"; fi
-@@KERNEL_BOOTCMD@@ ${kernel_addr_r} - ${fdt_addr}
+
+if env exists rpipart;
+then 
+  if test "${have_updated}" = 1; then
+    if test "${boot_attempts}" = "3";
+    then 
+      # do a reversion based upon rpipart
+      if test "${rpipart}" = "2"; then
+        # Revert to three
+        setenv rpipart 3
+        setenv boot_attempts 0
+      else
+        if test "${rpipart}" = "3"; then
+          # Revert to two
+          setenv rpipart 2
+          setenv boot_attempts 0
+        fi
+      fi
+
+    fi
+  fi
+  
+  setenv bootargs "${bootargs} root=/dev/mmcblk0p${rpipart}"; 
```

Comment:
should the mmcblk0 prefix of the device file also be parameterized as a u-boot env variable and set via fw setenv in non-boot user-land? The way it is currently is it only supports SD card storage on PCI.
For instance, it would be mmcblk0 for PCI sd card storage, but internal usb card reader could be bound to, say, /dev/sda1 parameterized as /dev/${blkdev}${rpipart}